### ISSUE TITLE
iOS Voice Control does not work for nested text inputs

### ISSUE THREAD

**Issue by dcalhoun** [WordPress/gutenberg#45415]
## Description

iOS [Voice Control](https://support.apple.com/en-us/HT210417) commands to not function properly when focusing a nested text input, e.g. a Paragraph within a Columns block. This was originally reported in https://github.com/wordpress-mobile/WordPress-iOS/issues/19175, which included additional contexts where Voice Control failed. 

## Step-by-step reproduction instructions

1. Activate iOS [Voice Control](https://support.apple.com/en-us/HT210417) on your iOS device.
1. Add a Columns block. 
2. Add a Paragraph block within the first column. 
3. Add text to the Paragraph block. 
1. Dictate "move to beginning."
1. Dictate "move to end."

## Expected behaviour

The cursor position is moved the beginning and then to the end based upon the voice commands. 

## Actual behaviour

The cursor position does not update. 

## Screenshots or screen recording (optional)

<details><summary>Screencast</summary>

https://user-images.githubusercontent.com/438664/199005489-0f0e5db8-c196-4301-83cc-29d5ca862fc2.MP4

</details>

## WordPress information
- WordPress version:  6.0
- Gutenberg version:  14.4.0
- Are all plugins except Gutenberg deactivated?  Yes
- Are you using a default theme (e.g. Twenty Twenty-One)?  Geologist

## Device information
- Device:  iPhone SE
- Operating system:  16.1 
- WordPress app version:  21.0.0.1
